### PR TITLE
cleanup hostinfo

### PR DESCRIPTION
**Pull request by rphillips**
* Remove BuildResult
* Use Asserts in unit tests

### CODE REVIEW

File: hostinfo/hostinfo.go [hunk 1]
```diff
@@ -18,18 +18,16 @@
...
-	Run() (*check.ResultSet, error)
-	BuildResult(cr *check.ResultSet) interface{}
+	Run() (interface{}, error)
 }
 
-func NewHostInfoResponse(cr *check.ResultSet, f *protocol.FrameMsg, hinfo HostInfo) *protocol.HostInfoResponse {
+func NewHostInfoResponse(result interface{}, f *protocol.FrameMsg) *protocol.HostInfoResponse {
 	resp := &protocol.HostInfoResponse{}
-	resp.Result = hinfo.BuildResult(cr)
+	resp.Result = result
```
Comment (itzg) [inline hostinfo/hostinfo.go]: Oh...because json.Marshal can "see into" interface/pointers? Awesome!

File: hostinfo/hostinfo_cpu.go [hunk 2]
```diff
@@ -46,59 +44,25 @@ func (*HostInfoCpu) Run() (*check.ResultSet, error) {
...
-		metrics.Wait, _ = cr.GetMetric("wait").ToFloat64()
-		idle, _ := cr.GetMetric("idle").ToFloat64()
-		metrics.Idle = uint64(idle)
-		total, _ := cr.GetMetric("total").ToFloat64()
-		metrics.Total = uint64(total)
-		metrics.Stolen, _ = cr.GetMetric("stolen").ToFloat64()
-		metrics.Irq, _ = cr.GetMetric("irq").ToFloat64()
-		metrics.Sys, _ = cr.GetMetric("sys").ToFloat64()
-		metrics.Mhz, _ = cr.GetMetric("mhz").ToFloat64()
+		metrics.Name = fmt.Sprintf("cpu.%d", i)
```
Comment (itzg) [inline hostinfo/hostinfo_cpu.go]: So much nicer

File: hostinfo/hostinfo_filesystem_test.go [hunk 3]
```diff
@@ -98,161 +96,19 @@ func TestHostInfoFilesystem_Run(t *testing.T) {
 			h := &hostinfo.HostInfoFilesystem{
 				HostInfoBase: protocol_hostinfo.HostInfoBase{},
 			}
-			got, err := h.Run()
+			got, _ := h.Run()
 			if tt.expectedErr {
 				assert.Error(t, err, fmt.Sprintf("HostInfoFilesystem.Run() error = %v, expectedErr %v", err, tt.expectedErr))
```
Comment (itzg) [inline hostinfo/hostinfo_filesystem_test.go]: Is the right `err` to be asserting?

File: hostinfo/hostinfo_memory.go [hunk 4]
```diff
@@ -48,42 +48,16 @@ func (*HostInfoMemory) Run() (*check.ResultSet, error) {
...
-	result.Metrics.SwapTotal, _ = cr.GetMetric("SwapTotal").ToUint64()
-	result.Metrics.SwapUsed, _ = cr.GetMetric("SwapUsed").ToUint64()
-	result.Metrics.SwapUsedPercentage, _ = cr.GetMetric("SwapUsedPercentage").ToFloat64()
-	ram, _ := cr.GetMetric("Total").ToUint64()
-	result.Metrics.RAM = uint64(ram / Megabyte)
-
-	return result
+	result.Metrics.UsedPercentage = v.UsedPercent
+	result.Metrics.Free = v.Free
+	result.Metrics.ActualFree = v.Free
```
Comment (itzg) [inline hostinfo/hostinfo_memory.go]: Just double checking that there isn't enough granularity to pull an "actual" value?

File: hostinfo/hostinfo_memory_test.go [hunk 5]
```diff
@@ -86,143 +83,24 @@ func TestHostInfoMemory_Run(t *testing.T) {
...
-			expected: &protocol_hostinfo.HostInfoMemoryResult{
-				Timestamp: utils.NowTimestampMillis(),
-			},
-		},
-	}
-	for _, tt := range tests {
-		t.Run(tt.name, func(t *testing.T) {
-			h := &hostinfo.HostInfoMemory{
-				HostInfoBase: protocol_hostinfo.HostInfoBase{},
+				//for name, sample_metric := range tt.expected {
```
Comment (itzg) [inline hostinfo/hostinfo_memory_test.go]: Could just delete these commented lines until new style asserts are in place.

File: hostinfo/hostinfo_processes_test.go [hunk 6]
```diff
@@ -159,22 +159,12 @@ func TestHostInfoProcesses_Run(t *testing.T) {
 			h := &hostinfo.HostInfoProcesses{
 				HostInfoBase: protocol_hostinfo.HostInfoBase{},
 			}
-			got, err := h.Run()
+			got, _ := h.Run()
 			if tt.expectedErr {
 				assert.Error(t, err, fmt.Sprintf("HostInfoProcesses.Run() error = %v, expectedErr %v", err, tt.expectedErr))
```
Comment (itzg) [inline hostinfo/hostinfo_processes_test.go]: Same question, asserting the right `err`?

File: hostinfo/hostinfo_system_test.go [hunk 7]
```diff
@@ -113,119 +112,18 @@ func TestHostInfoSystem_Run(t *testing.T) {
 			h := &hostinfo.HostInfoSystem{
 				HostInfoBase: protocol_hostinfo.HostInfoBase{},
 			}
-			got, err := h.Run()
+			got, _ := h.Run()
 			if tt.expectedErr {
 				assert.Error(t, err, fmt.Sprintf("HostInfoSystem.Run() error = %v, expectedErr %v", err, tt.expectedErr))
```
Comment (itzg) [inline hostinfo/hostinfo_system_test.go]: Same... :)

File: hostinfo/hostinfo_filesystem_test.go (hunk 3 above)
Comment (rphillips) [inline hostinfo/hostinfo_filesystem_test.go]: fixed in 46747f5

File: hostinfo/hostinfo_memory.go (hunk 4 above)
Comment (rphillips) [inline hostinfo/hostinfo_memory.go]: There isn't `actual` in gops yet... we may want to eventually add it.

File: hostinfo/hostinfo_memory_test.go (hunk 5 above)
Comment (rphillips) [inline hostinfo/hostinfo_memory_test.go]: sorry, missed this one. fixed in cc7a748

File: hostinfo/hostinfo_processes_test.go (hunk 6 above)
Comment (rphillips) [inline hostinfo/hostinfo_processes_test.go]: fixed

**Comment by rphillips** [comment]
thanks... should be good to go now.

**Comment by itzg** [comment]
+1

**Pull request merged**